component/upgrade/patch-crds.sh

```diff
@@ -6,7 +6,7 @@ cd /export
 for crd in ${CRDS_TO_PATCH}; do
   # Export and remove all metadata properties except `metadata.name`
   kubectl get crd "${crd}" -o json > "${crd}.json"
-  jq 'del(.status) | del(.metadata) | .metadata.name += "'${crd}'"' "${crd}.json" > "${crd}.patched.json"
+  jq 'del(.status,.metadata.resourceVersion,.metadata.uid,.metadata.annotations,.metadata.creationTimestamp,.metadata.selfLink)' "${crd}.json" > "${crd}.patched.json"
```

Comment:
I wouldn't delete the annotations here. Maybe it may be necessary to delete the kubectl-last applied annotation, but there could be other annotations we don't know.

I think `managedFields` should also be deleted

Reply:
copypasta error 🙈

Reply:
See here which fields are part of metadata: https://github.com/kubernetes/apimachinery/blob/ff522ab81c745a9ac5f7eeb7852fac134194a3b6/pkg/apis/meta/v1/types.go#L116

Reply:
Basically delete everything except `name`, `labels`,`annotations` I'd say

Reply:
> Basically delete everything except `name`, `labels`,`annotations` I'd say

Would you be ok with the following filter?

```sh
jq 'del(.status) | .metadata = (.metadata | {name, labels, annotations})'
```

It removes everything in `metadata` except `name, labels, annotations` even if future k8s versions add new fields.

Reply:
yes ofc, didn't know this syntax, I'm no jq/yq expert :)

Reply:
needed one or two tries too 😅